src/plugins/orangehrmBuzzPlugin/Dao/BuzzDao.php

```diff
@@ -414,4 +414,22 @@ public function getBuzzShareByPostId(int $postId): ?BuzzShare
 
         return $q->getQuery()->getOneOrNullResult();
     }
+
+    /**
+     * @param int $shareId
+     * @return BuzzShare[]
+     */
+    public function getBuzzPostSharesById(int $shareId): array
+    {
+        /** @var BuzzShare $share */
+        $share = $this->getRepository(BuzzShare::class)->find($shareId);
+
+        if ($share->getType() === BuzzShare::TYPE_SHARE) {
+            return [];
+        }
+
+        return $this->getRepository(BuzzShare::class)->findBy(
+            ['post' => $share->getPost()->getId(), 'type' => BuzzShare::TYPE_SHARE]
+        );
+    }
```

Comment:
@RajithaKumara please confirm this behaviour. if the shared post is a reshare (type `TYPE_SHARE = 1`) then an empty array will be returned. Only for the original post will shares be returned.

There is no way to check if a reshare has been shared again since it links back to the original post in the table

Reply:
I did some changes to feed API as well;
https://github.com/orangehrm/orangehrm/pull/1558/commits/fa6e8327544bb45088763a5050709678d9aab179

```sql
select count(BuzzShare.id) from BuzzShare where BuzzShare.type = BuzzShare::TYPE_SHARE and BuzzShare.id = :shareId
--- OR
select count(BuzzShare.id) from BuzzShare where BuzzShare.type = BuzzShare::TYPE_SHARE and IDENTITY(BuzzShare.post) = :postId
```